driver-sync/src/test/functional/com/mongodb/client/AbstractClientSideOperationsTimeoutProseTest.java

```diff
@@ -0,0 +1,236 @@
+ */
+
+package com.mongodb.client;
+
+import com.mongodb.ClusterFixture;
+import com.mongodb.ConnectionString;
+import com.mongodb.CursorType;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.MongoExecutionTimeoutException;
+import com.mongodb.MongoNamespace;
+import com.mongodb.MongoTimeoutException;
+import com.mongodb.ReadConcern;
+import com.mongodb.ReadPreference;
+import com.mongodb.WriteConcern;
+import com.mongodb.client.model.CreateCollectionOptions;
+import com.mongodb.client.model.changestream.ChangeStreamDocument;
+import com.mongodb.client.test.CollectionHelper;
+import com.mongodb.event.CommandEvent;
+import com.mongodb.internal.connection.ServerHelper;
+import com.mongodb.internal.connection.TestCommandListener;
+import com.mongodb.test.FlakyTest;
+import org.bson.BsonDocument;
+import org.bson.BsonInt32;
+import org.bson.BsonTimestamp;
+import org.bson.Document;
+import org.bson.codecs.BsonDocumentCodec;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Named;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import static com.mongodb.ClusterFixture.isDiscoverableReplicaSet;
+import static com.mongodb.ClusterFixture.isStandalone;
+import static com.mongodb.ClusterFixture.serverVersionAtLeast;
+import static com.mongodb.ClusterFixture.sleep;
+import static com.mongodb.client.Fixture.getDefaultDatabaseName;
+import static com.mongodb.client.Fixture.getPrimary;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+/**
+ * See
+ * <a href="https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/tests/README.rst">Prose Tests</a>.
+ */
+public abstract class AbstractClientSideOperationsTimeoutProseTest {
+
+    private final MongoNamespace namespace = new MongoNamespace(getDefaultDatabaseName(), this.getClass().getSimpleName());
+    private TestCommandListener commandListener;
+    private CollectionHelper<BsonDocument> collectionHelper;
+
+    protected abstract MongoClient createMongoClient(MongoClientSettings mongoClientSettings);
+
+    @Tag("setsFailPoint")
+    @FlakyTest(maxAttempts = 3)
+    @DisplayName("5. Blocking Iteration Methods - Tailable cursors")
+    public void testBlockingIterationMethodsTailableCursor() {
+        assumeTrue(serverVersionAtLeast(4, 4));
+        assumeTrue(isStandalone());
```

Comment:
I failed to get it to pass on evergreen with replicasets. Perhaps with read and write concern majority the timeouts are just too small.